src/entry.rs

```diff
@@ -659,12 +677,12 @@ impl<'a> Serialize for EntryItem<'a> {
                 } else if let Some(s) = self.body_as_struct_array::<memory::Ddr5CaPinMapElement>() {
                     let v = s.iter().collect::<Vec<_>>();
                     state.serialize_field("Ddr5CaPinMapElement", &v)?;
-                } else if let Some(s) = self.body_as_struct_array::<memory::MemDfeSearchElement20>() {
-                    let v = s.iter().collect::<Vec<_>>();
-                    state.serialize_field("MemDfeSearchElement20", &v)?;
-                } else if let Some(s) = self.body_as_struct_array::<memory::MemDfeSearchElement32>() {
+//                } else if let Some(s) = self.body_as_struct_array::<memory::MemDfeSearchElement32>() { // UH OH
```

Comment:
Yeah. There seem to be two different versions of that struct MemDfeSearchElement in use (the number is the size in bytes in decimal).

Since those structs are used in a struct array and there's only a `total_size` field somewhere it could (and did) happen that it's misdetected since it's possible that `sizeof(MemDfeSearchElement32)` divides `total_size` and also `sizeof(MemDfeSearchElement36)` divides `total_size` at the same time.

Reply:
Analysis of what we have: 19 of 54 directories in fluffy-tribble do not have that struct size = 36.

Details: Only the following fail assert sizeof == 36:

```
./am5-1.0.0.6/AgesaPkg/Addendum/Apcb/RaphaelAm5/ApcbDataDefaultRecovery/CRB_Splinter
./am5-1.0.0.6/AgesaPkg/Addendum/Apcb/RaphaelAm5Alt1/ApcbDataDefaultRecovery/CRB_Shredder
./genoa-1.0.0.3/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/GenoaCommon
./genoa-1.0.0.3/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/Ruby
./genoa-1.0.0.9/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/GenoaCommon
./genoa-1.0.0.9/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/Cinnabar
./genoa-1.0.0.9/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/Ruby
./genoa-1.0.0.9/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/Sunstone
./genoa-1.0.0.a/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/GenoaCommon
./genoa-1.0.0.a/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/Cinnabar
./genoa-1.0.0.a/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/Ruby
./genoa-1.0.0.a/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/Ruby
./genoa-1.0.0.a/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/Sunstone
./genoa-1.0.0.b/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/GenoaCommon
./genoa-1.0.0.b/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/Cinnabar
./genoa-1.0.0.b/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/Ruby
./genoa-1.0.0.b/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/Sunstone
./genoa-1.0.0.8/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/GenoaCommon
./genoa-1.0.0.8/AgesaPkg/Addendum/Apcb/GenoaSp5Rdimm/ApcbDataDefaultRecovery/Ruby
```